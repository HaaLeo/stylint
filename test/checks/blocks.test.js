'use strict';

const blockTest = require('../../src/checks/blocks');

describe('blocks', () => {
  describe('prefer @block when defining block vars', () => {
    let context;

    beforeEach(() => {
      context = {
        report: jest.fn(),
        conf: 'always',
      };
    });

    it('report if block style incorrect', () => {
      blockTest(Object.assign({}, context, { line: 'myBlock = ' }));
      blockTest(Object.assign({}, context, { line: 'myBlock =' }));

      expect(context.report.mock.calls).toMatchSnapshot();
    });

    it("don't report if block style correct", () => {
      blockTest(Object.assign({}, context, { line: 'myBlock = @block' }));
      blockTest(Object.assign({}, context, { line: 'myBlock = @block ' }));

      expect(context.report).not.toHaveBeenCalled();
    });

    it("don't report if block style not applicable", () => {
      blockTest(Object.assign({}, context, { line: '.class' }));
      blockTest(Object.assign({}, context, { line: 'input[type="submit"]' }));

      expect(context.report).not.toHaveBeenCalled();
    });
  });

  describe('disallow @block when defining block vars', () => {
    let context;

    beforeEach(() => {
      context = {
        report: jest.fn(),
        conf: 'never',
      };
    });

    it("don't report if block style IS correct", () => {
      blockTest(Object.assign({}, context, { line: 'myBlock = ' }));
      blockTest(Object.assign({}, context, { line: 'myBlock =' }));

      expect(context.report).not.toHaveBeenCalled();
    });

    it('report if block style NOT correct', () => {
      blockTest(Object.assign({}, context, { line: 'myBlock = @block' }));
      blockTest(Object.assign({}, context, { line: 'myBlock = @block ' }));

      expect(context.report.mock.calls).toMatchSnapshot();
    });

    it("don't report if block style not applicable", () => {
      blockTest(Object.assign({}, context, { line: '.class' }));
      blockTest(Object.assign({}, context, { line: 'input[type="submit"]' }));

      expect(context.report).not.toHaveBeenCalled();
    });
  });
});
