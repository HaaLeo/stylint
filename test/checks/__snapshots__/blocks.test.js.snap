// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks disallow @block when defining block vars report if block style NOT correct 1`] = `
Array [
  Array [
    Object {
      "column": 10,
      "message": "@block is not allowed",
    },
  ],
  Array [
    Object {
      "column": 10,
      "message": "@block is not allowed",
    },
  ],
]
`;

exports[`blocks prefer @block when defining block vars report if block style incorrect 1`] = `
Array [
  Array [
    Object {
      "column": 10,
      "message": "block variables must include @block",
    },
  ],
  Array [
    Object {
      "column": 9,
      "message": "block variables must include @block",
    },
  ],
]
`;
